src/sagemaker/image_uris.py

```diff
@@ -463,6 +531,25 @@ def _validate_version_and_set_if_needed(version, config, framework):
         return available_versions[0]
 
     _validate_arg(version, available_versions + aliased_versions, "{} version".format(framework))
+
+    # For DLCs, warn if image is out of support.
+    eos_version, eos_config, eos_framework = version, config, framework
+
+    # Default to underlying base framework for HF, except for neuron which has its own policies.
+    inf_or_trn = "inf" in config.get("processors", []) or "trn" in config.get("processors", [])
```

Comment:
This does mean that HF will need to keep up-to-date with latest patch version releases of frameworks to stay in support